### PR TITLE
Adds Detail Screen

### PR DESCRIPTION
**Pull request by jjszolno**
### Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

Fixes Navigation and shows detail screen with images carousel.

---

#### Jira board reference

- [Product Detail](https://www.notion.so/rootstrap/Product-detail-45951eb36ee5414389921d77a005fca9?pvs=4)

---

#### Notes:

-

---

#### Tasks:

- [ ] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [x] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

### CODE REVIEW

File: src/navigation/stacks/main/index.tsx [hunk 1]
```diff
@@ -59,6 +82,32 @@ const MainStack = () => {
...
+      <StackNav.Screen name={MainStackScreens.Detail} component={DetailScreen} />
+    </StackNav.Navigator>
+  );
+};
+
+const HomeStack = () => {
+  return (
+    <StackNav.Navigator screenOptions={{ headerShown: false }}>
+      <StackNav.Screen name={MainStackScreens.Home} component={HomeScreen} />
+      <StackNav.Screen name={MainStackScreens.Detail} component={DetailScreen} />
```
Comment (guillermomachado) [inline src/navigation/stacks/main/index.tsx:100]: You're including this screen twice, in `ProductsStack` and here, is this needed?

File: src/features/products/screen/productItem.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    flexDirection: 'row',
+    borderBottomWidth: 1,
+    borderBlockColor: '#000000',
+  },
+  containerLast: {
+    height: 150,
+    width: '100%',
+    padding: 10,
+    flexDirection: 'row',
+  },
```
Comment (guillermomachado) [inline src/features/products/screen/productItem.tsx]: You may want to apply only the different styles to the last container. Remember the `style` prop accepts an array of styles, so you could do: `[styles.container, isLast ? styles.containerLast : ...[]]`

Comment (guillermomachado) [reply src/features/products/screen/productItem.tsx]: Not blocker tho

File: src/features/productDetail/screen/selector.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  const handleQuantityChange = (itemValue: string) => {
+    setSelectedQuantity(itemValue);
+  };
+
+  return (
+    <Picker selectedValue={selectedQuantity} onValueChange={handleQuantityChange}>
+      {Array.from({ length: 10 }, (_, index) => (
+        <Picker.Item key={index + 1} label={`${index + 1}`} value={`${index + 1}`} />
+      ))}
```
Comment (guillermomachado) [inline src/features/productDetail/screen/selector.tsx:16]: Smart

**Review by guillermomachado — APPROVED** [review]

File: src/navigation/stacks/main/index.tsx (hunk 1 above)
Comment (jjszolno) [reply src/navigation/stacks/main/index.tsx:100]: It was needed. The app didnt know where to go if I didnt add it there

Comment (guillermomachado) [reply src/navigation/stacks/main/index.tsx:100]: If they live under the same main route, you could do the following (from the Home Stack):

`navigate(ProductStack, { screen: DetailsScreen, params: { productId: abc123 } })`

**Pull request merged**